packages/myst-to-react/src/links/index.tsx

```diff
@@ -134,8 +137,20 @@ export const linkBlock: NodeRenderer<TransformedLink> = ({ node }) => {
   );
 };
 
-const LINK_RENDERERS = {
-  link,
+const LINK_RENDERERS: NodeRenderers = {
+  link: {
```

Comment:
I don't quite see why we need this nested structure as opposed to:
```ts
{
  link: SimpleLink,
  'link[protocol=github]': GithubLinkRenderer,
  'link[protocol=wiki]': WikiLinkRenderer,
  ...
}
```

I guess the nesting lets us first narrow down based on `type` so we are not doing `selector` checks against _all_ the `renderers`?

Reply:
My next question is: would this work the same if we did something like:
```ts
{
  link: {
    base: SimpleLink,
    '[protocol=github]': GithubLinkRenderer,
    '[protocol=wiki]': WikiLinkRenderer,
    ...
  }
}
```

i.e. we don't need to repeat `"link"` since we are only dealing with `link` nodes at this point?